src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -12131,5 +12131,512 @@ public string Prop
+        {
+            var source = """
+                #nullable enable
+
+                public struct S()
+                {
+                    public string Prop1
+                    {
+                        get
+                        {
+                            Prop2 = "a";
+                            this = default;
+                            Prop2.ToString(); // expected warning missing
+                            StaticProp.ToString();
+                            return "a";
+                        }
+                    }
+
+                    public string Prop2
+                    {
+                        get => field;
+                        set => field = value;
+                    } = "b";
+
+                    public static string StaticProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = "c";
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/77991")]
+        public void Nullable_PropInitializer_PropStateAffectedByAssigningStruct()
+        {
+            var source = """
+                #nullable enable
+
+                public struct S()
+                {
+                    public string Prop { get => field; set => field = value; } = "a";
+
+                    public static S StaticProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = M(
+                        StaticProp = default,
+                        StaticProp.Prop.ToString()); // expected warning missing
+
+                    public static S OtherProp
+                    {
+                        get => field;
+                        set => field = value;
+                    } = M(
+                        StaticProp = default,
+                        StaticProp.Prop.ToString()); // expected warning missing
+
+                    public static S M(S s, string str) => s;
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Nullable_Cycle_02()
+        {
+            var source = """
+                #nullable enable
+
+                var s = new S
+                {
+                    Resilient = "a", // 1
+                    NonResilient = "b" // 2
+                };
+
+                public struct S
+                {
+                    public S()
+                    {
+                        Resilient.ToString(); // 3
+                        NonResilient.ToString(); // 4
+                    }
+
+                    public S(bool ignored) : this()
+                    {
+                        Resilient.ToString(); // 5
```

Comment:
>Resilient.ToString(); // 5 [](http://example.com/codeflow?start=24&length=26)

Why are we reporting a warning here, rather than assuming the chained constructor initialized the property?

Reply:
This is the main scenario in bug #77991. The main reason is that the code in GetOrCreateSlot is using the field to "own" the shared slot in the context of the constructor. The field is therefore also used to give the slot its initial flow state (`GetDefaultState()`). That results in maybe-null state for the resilient field as a maybe-null annotation was inferred for it. And a not-null state for the other field since not-null was inferred for it.

Reply:
I think it will be good to come back and iron this out more completely for a future release, but for now, to just plug the gap which is resulting in a crash, and handle other possible cycles better at runtime.

Reply:
No I think #77991 is adequate as a catch all, I'll ensure the tests with unexpected/missing warnings are all marked.